libethash-cuda/ethash_cuda_miner_kernel.cu

```diff
@@ -45,18 +41,38 @@ void run_ethash_search(
 	uint32_t sharedbytes,
 	cudaStream_t stream,
 	volatile uint32_t* g_output,
-	uint64_t start_nonce
+	uint64_t start_nonce,
+        uint32_t parallelHash
 )
 {
-	ethash_search << <blocks, threads, sharedbytes, stream >> >(g_output, start_nonce);
+
+        //printf("parallelHash = %d\n", parallelHash);
+        if(parallelHash == 1)
```

Comment:
I used to use switch, but met some issues with template, so I changed switch to if..else if...else if.
I will try switch again.